packages/component-utilities/src/getCompletedData.js

```diff
@@ -1,132 +1,83 @@
-	let min;
-	let max;
-	if (valueof === undefined) {
-		for (const value of values) {
-			if (value != null) {
-				if (min === undefined) {
-					if (value >= value) min = max = value;
-				} else {
-					if (min > value) min = value;
-					if (max < value) max = value;
-				}
-			}
-		}
-	} else {
-		let index = -1;
-		for (let value of values) {
-			if ((value = valueof(value, ++index, values)) != null) {
-				if (min === undefined) {
-					if (value >= value) min = max = value;
-				} else {
-					if (min > value) min = value;
-					if (max < value) max = value;
-				}
-			}
+import { findInterval, vectorSeq } from './helpers/getCompletedData.helpers.js';
+
+/**
+ * This function fills missing data points in the given data array for a specific series.
+ *
+ * @param {Array<object>} data - The data as an array of objects.
+ * @param {string} x - The property used as x-axis.
+ * @param {string} y - The property used as y-axis.
+ * @param {string} series - The specific series in the data to be filled.
+ * @param {boolean} [nullsZero=false] - A flag indicating whether nulls should be replaced with zero.
+ * @param {boolean} [fillX=false] - A flag indicating whether the x-axis values should be filled (based on the found interval distance).
+ * @return {Array<object>} An array containing the filled data objects.
+ */
+export default function getCompletedData(data, x, y, series, nullsZero = false, fillX = false) {
+	const groups = Array.from(data).reduce((a, v) => {
+		if (series) {
+			if (!a[v[series]]) a[v[series]] = [];
+			a[v[series]].push(v);
+		} else {
+			if (!a.default) a.default = [];
+			a.default.push(v);
 		}
-	}
-	return [min, max];
-}
-
-function vectorSeq(values, period) {
-	let min = extent(values)[0];
-	let max = extent(values)[1];
-
-	const sequence = [];
-	let value = min;
-	while (value <= max) {
-		sequence.push(Math.round((value + Number.EPSILON) * 100000000) / 100000000);
-		value += period;
-	}
-
-	return sequence;
-}
-
-function findInterval(arr) {
-	if (arr.length === 1) {
-		return;
-	}
-	// Sort array ascending
-	arr.sort(function (a, b) {
-		return a - b;
-	});
-
-	// 1. Multiply array by 100
-	arr = arr.map(function (x) {
-		return x * 100000000;
-	});
+		return a;
+	}, {});
 
-	// 2. Get diffs
-	arr = getDiffs(arr);
+	const output = [];
+	for (const [key, value] of Object.entries(groups)) {
+		/** @type {Array<number>} */
+		let xDistinct;
 
-	// 3. Calculate greatest common divisor of diffs and divide by 100
-	let interval = arr.reduce(gcd) / 100000000;
-	interval = Math.round((interval + Number.EPSILON) * 100000000) / 100000000;
+		let xIsDate = false;
 
-	return interval;
-}
-
-export default function getCompletedData(data, x, y, series, nullsZero = false, fillX = false) {
-	let xDistinct = getDistinctValues(data, x);
-	let interval;
-	let filledData;
+		if (value[0]?.[x] instanceof Date) {
+			xDistinct = getDistinctValues(
+				value.map((d) => ({ ...d, [x]: d[x].getTime() })),
```

Comment:
```suggestion
				value.map((d) => ({ [x]: d[x].getTime() })),
```
Only `x` is accessed in the created array so shouldn't bother spreading `d`